server/netop_grpc_server.go

```diff
@@ -37,7 +36,7 @@ func (s *NetOps) UpdateSliceQosProfile(ctx context.Context, qosProfile *netops.S
 	if ctx.Err() == context.Canceled {
 		return nil, status.Errorf(codes.Canceled, "Client canceled, ignoring qos update message.")
 	}
-	if qosProfile == nil {
+	if qosProfile.SliceId == "" {
```

Comment:
Shouldnt this be a separate check with a different error message?

Reply:
@bharath-avesha , bufconn (the package which was used to start a grpc server) was unable to unmarshal pointer types and for this reason in order to validate if a Qos Profile is empty or not i am checking if the primary fields of the struct are empty or not

Reply:
Hmmm ok. But does that mean the qosProfile will never be null? Since the callback has one of the parameters as a pointer type, i am wondering how the unmarshalling component would populate data for that parameter.

Reply:
@bharath-avesha yeah it is never nil, its always an empty object something like this: `&netops.SliceQosProfile{SliceName:"", SliceId:"", QosProfileName:"", TcType:0, ClassType:0, BwCeiling:0x0, BwGuaranteed:0x0, Priority:0x0, DscpClass:"", XXX_NoUnkeyedLiteral:struct {}{}, XXX_unrecognized:[]uint8(nil), XXX_sizecache:0}`